File: renderer/src/main/java/io/sparkled/renderer/context/RenderContext.java
package io.sparkled.renderer.context;

import io.sparkled.model.animation.effect.Effect;
import io.sparkled.model.entity.Song;
import io.sparkled.model.render.RenderedStagePropData;
import io.sparkled.model.render.RenderedFrame;

public class RenderContext {
    private final Song song;
    private final RenderedStagePropData channel;
    private final RenderedFrame frame;
    private final Effect effect;
    private final float progress;

    public RenderContext(Song song, RenderedStagePropData channel, RenderedFrame frame, Effect effect, float progress) {
        this.song = song;
        this.channel = channel;
        this.frame = frame;
        this.effect = effect;
        this.progress = progress;
    }

    public Song getSong() {
        return song;
    }

    public RenderedStagePropData getChannel() {
        return channel;
    }

    public RenderedFrame getFrame() {
        return frame;
    }

    public Effect getEffect() {
        return effect;
    }

    public float getProgress() {
        return progress;
    }
}

File: renderer/src/main/kotlin/io/sparkled/renderer/context/RenderContext.kt
package io.sparkled.renderer.context

import io.sparkled.model.animation.effect.Effect
import io.sparkled.model.entity.Sequence
import io.sparkled.model.render.RenderedFrame
import io.sparkled.model.render.RenderedStagePropData

data class RenderContext(
    val sequence: Sequence,
    val channel: RenderedStagePropData,
    val frame: RenderedFrame,
    val effect: Effect,
    val progress: Float
)
